driver-core/src/test/java/com/datastax/driver/core/CCMBridge.java

```diff
@@ -1332,6 +1314,21 @@ private String randomizePorts(CharSequence str) {
       return sb.toString();
     }
 
+    // Collects port numbers from configuration map. May catch some non-issue false positives.
+    private HashSet<Integer> collectPortNumbers(Map<String, Object> map, HashSet<Integer> set) {
+      for (Map.Entry<String, Object> entry : map.entrySet()) {
+        Object value = entry.getValue();
+        try {
+          Integer result = Integer.parseInt(value.toString());
+          if ((1 <= result) && (result <= 65535)) {
+            set.add(result);
+          }
+        } catch (NumberFormatException e) {
+        }
+      }
+      return set;
```

Comment:
I think it would be cleaner if the signature of the method was `private static Collection<Integer> collectPorts(Map<String, Object> config) {` and the method internally creates a Set, populates it and returns. Then the usage would be `selectedPorts.addAll(collectPorts(this.dseConfiguration))`. Notice the change in name that drops Numbers suffix.

Reply:
I've changed it as suggested.